### PR TITLE
NO-JIRA-YET: Set the buffer size to 4k. Also adjusts Q2 and Q3 accord…

### PR DESCRIPTION
**Pull request by ganeshmurthy**
…ingly.

### CODE REVIEW

File: src/buffer.c [hunk 1]
```diff
@@ -24,8 +24,8 @@
 #include <stdint.h>
 #include <string.h>
 
-
-size_t BUFFER_SIZE     = 512;
+// BUFFER_SIZE is set to 4k
```
Comment (ChugR) [inline src/buffer.c:27]: "size_t BUFFER_SIZE = 4096;" 

stands pretty well on its own and does not need a comment.

**Review by ChugR — APPROVED** [review]
This patch looks OK. It would be interesting to verify that 512 vs 4096 router versions interoperate well.

File: src/buffer.c (hunk 1 above)
Comment (ganeshmurthy) [reply src/buffer.c]: Removed comment, thanks

File: tests/run_unit_tests_size.c [hunk 2]
```diff
@@ -45,7 +45,7 @@ int main(int argc, char** argv)
     qd_buffer_set_size(buffer_size);
 
     int result = 0;
-    result += message_tests();
+    result += message_tests(buffer_size);
```
Comment (kgiusti) [inline tests/run_unit_tests_size.c:48]: should the default buffer size be updated on line 34 above?

**Comment by kgiusti** [comment]
There's some test failures that might be relying on the 512 buffer size.

https://github.com/apache/qpid-dispatch/blob/main/tests/system_tests_one_router.py#L3577

Also we need to add a 4096 test case here:
https://github.com/apache/qpid-dispatch/blob/main/tests/CMakeLists.txt#L76

**Review by kgiusti — COMMENTED** [review]
I would also define python constants in tests/system_test.py for BUFFER_SIZE and all the Q2 and Q3 values in message.h.   These can be used by the tests for testing buffer/Q2/Q3 limits.

I'd also add a big comment around the C definitions that says if these values change the corresponding values in tests/system_test.py must be updated.

**Comment by kgiusti** [comment]
Oooh boy - HTTP1 tests don't see to be too happy with these changes...

**Comment by ganeshmurthy** [comment]
> Oooh boy - HTTP1 tests don't see to be too happy with these changes...

I saw several "OSError: [Errno 98] Address already in use" errors and assumed those had to do with https://issues.apache.org/jira/browse/DISPATCH-1986

**Pull request closed without merging**